xbmc/FileItem.cpp

```diff
@@ -339,61 +339,20 @@ const CFileItem& CFileItem::operator=(const CFileItem& item)
   m_bIsShareOrDrive = item.m_bIsShareOrDrive;
   m_dateTime = item.m_dateTime;
   m_dwSize = item.m_dwSize;
-  if (item.HasMusicInfoTag())
-  {
-    m_musicInfoTag = GetMusicInfoTag();
-    if (m_musicInfoTag)
-      *m_musicInfoTag = *item.m_musicInfoTag;
-  }
-  else
-  {
-    delete m_musicInfoTag;
-    m_musicInfoTag = NULL;
-  }
-
-  if (item.HasVideoInfoTag())
-  {
-    m_videoInfoTag = GetVideoInfoTag();
-    if (m_videoInfoTag)
-      *m_videoInfoTag = *item.m_videoInfoTag;
-  }
-  else
-  {
-    delete m_videoInfoTag;
-    m_videoInfoTag = NULL;
-  }
 
-  if (item.HasEPGInfoTag())
-    m_epgInfoTag = item.m_epgInfoTag;
-  else
-    m_epgInfoTag.reset();
-
-  if (item.HasPVRChannelInfoTag())
-    m_pvrChannelInfoTag = item.m_pvrChannelInfoTag;
-  else
-    m_pvrChannelInfoTag.reset();
+  delete m_musicInfoTag;
+  m_musicInfoTag = item.m_musicInfoTag ? new MUSIC_INFO::CMusicInfoTag(*item.m_musicInfoTag) : NULL;
```

Comment:
why create a new instance in which we copy instead of just copying into it like it did? it's less code, sure, but this gets called a lot when creating file lists for the gui